cockatrice/src/game/player/player.cpp

```diff
@@ -3433,6 +3433,18 @@ void Player::actCardCounterTrigger()
     sendGameCommand(prepareGameCommand(commandList));
 }
 
+/**
+ * @brief returns true if the zone is a unwritable reveal zone view (eg a card reveal window)
+ */
+static bool isUnwritableRevealZone(CardZone *zone)
+{
+    if (zone && zone->getIsView()) {
+        auto *view = dynamic_cast<ZoneViewZone *>(zone);
```

Comment:
What's the purpose of dynamic casting here vs static casting?

Reply:
good point